engine/schema/src/test/java/com/cloud/upgrade/DatabaseUpgradeCheckerTest.java

```diff
@@ -74,6 +77,30 @@ public void testCalculateUpgradePath490to4910() {
 
     }
 
+    @Test
+    public void testCalculateUpgradePath410to412() {
+
+        final CloudStackVersion dbVersion = CloudStackVersion.parse("4.10.0.0");
+        assertNotNull(dbVersion);
+
+        final CloudStackVersion currentVersion = CloudStackVersion.parse("4.12.0.0");
+        assertNotNull(currentVersion);
+
+        final DatabaseUpgradeChecker checker = new DatabaseUpgradeChecker();
+        final DbUpgrade[] upgrades = checker.calculateUpgradePath(dbVersion, currentVersion);
+
+        assertNotNull(upgrades);
+        assertTrue(upgrades.length >= 1);
+        assertTrue(upgrades[0] instanceof Upgrade41000to41100);
+        assertTrue(upgrades[1] instanceof Upgrade41100to41110);
+        assertTrue(upgrades[2] instanceof Upgrade41110to41200);
+
+        assertTrue(Arrays.equals(new String[] { "4.11.0.0", "4.11.1.0"},
```

Comment:
@DaanHoogland, I think, Similarly we could add for upgrades[0] as well.

assertTrue(Arrays.equals(new String[] { "4.10.0.0", "4.11.0.0"}, upgrades[0].getUpgradableVersionRange()));

Reply:
it seem redundant to what is happening in other checks and in pther tests. Why do you think it is needed @nitin-maharana ?

Reply:
@DaanHoogland, you mean other tests to check upgrade path from 4.10 to 4.11? I just checked, no other tests handle this path.

Reply:
No i mean that the items are implicetly tested by the other checks in this test and also in other tests the completeness of the array is tested, therefore I am assuming it here.